app/models/assignment.rb

```diff
@@ -634,22 +634,26 @@ def summary_json(user)
 
   # Generates the summary of the most test results associated with an assignment.
   def summary_test_results
-    test_groups_query = self.test_groups
-                            .joins(test_group_results: { test_run: :grouping })
-                            .group('test_groups.id', 'groupings.id')
-                            .select('test_groups.id AS test_groups_id',
-                                    'MAX(test_group_results.created_at) AS test_group_results_created_at')
-                            .where.not('test_runs.submission_id': nil)
-                            .to_sql
-
-    self.test_groups
-        .joins(test_group_results: [:test_results, { test_run: { grouping: :group } }])
-        .joins("INNER JOIN (#{test_groups_query}) sub \
-                ON test_groups.id = sub.test_groups_id
-                AND test_group_results_test_groups.created_at = sub.test_group_results_created_at")
+    latest_test_run_by_grouping = TestRun.group('grouping_id').select('MAX(created_at) as test_runs_created_at',
+                                                                      'grouping_id')
+                                         .where.not(submission_id: nil)
+                                         .to_sql
+
+    latest_test_runs = TestRun
+                       .joins(grouping: :group)
+                       .joins("INNER JOIN (#{latest_test_run_by_grouping}) latest_test_run_by_grouping \
+            ON latest_test_run_by_grouping.grouping_id = test_runs.grouping_id \
+            AND latest_test_run_by_grouping.test_runs_created_at = test_runs.created_at")
+                       .select('id', 'test_runs.grouping_id', 'groups.group_name')
+                       .to_sql
+
+    self.test_groups.joins(test_group_results: :test_results)
+        .joins("INNER JOIN (#{latest_test_runs}) latest_test_runs \
+              ON test_group_results.test_run_id = latest_test_runs.id")
+        .joins('INNER JOIN groupings ON latest_test_runs.grouping_id = groupings.id')
```

Comment:
@pranavrao145 Oh wait, so actually now do we need a join on `groupings` at all? Seems like the only necessary information was to use the grouping to access the corresponding group name, but that's now been moved.

Reply:
Good point! No need to join it again, I will get rid of this one.